eng/common/TestResources/deploy-test-resources.yml

```diff
@@ -27,13 +42,31 @@ steps:
         ${{ parameters.SubscriptionConfiguration }}
       "@ | ConvertFrom-Json -AsHashtable;
 
+      if ($subscriptionConfiguration.ArmTemplateParameters) {
+        # Handle subscription configuration containing arm template parameters
+        $armTemplateParameters = $subscriptionConfiguration.ArmTemplateParameters
+      } else {
+        # Handle legacy subscription configuration, with arm template params passed in by templates
+        $armTemplateParameters = ${{ parameters.ArmTemplateParameters }}
+      }
+
+      $accessParameters = @{
```

Comment:
What is the point of copying this over? Is it to remove ArmTemplateParameters? If so I think it would be better to just go ahead and pass those as well. I believe they will be ignored for now but we could eventually stop explicitly passing AdditionalParameters and start consuming them from here.

Reply:
We chatted offline, but for context: the New-TestResources.ps1 treats the new fields as unknown parameters when they are splatted in as arguments. We decided to add all the parameters as bindings to the script, since eventually we'll want to handle all of them there.